Fix building on OpenGL ES 2.0 (RPI VideoCore IV)

joolswills opened the pull request:
GL_COMPRESSED_RGBA_ASTC_4x4_KHR is not defined and looks to be an OpenGL ES 3.0 extension.

This fixes building on the Raspberry Pi 1-3 using VideoCore IV OpenGL ES 2.0 headers.

joolswills commented:
Not sure if this is the right fix, so please feel free to comment/change/disregard. But GL_COMPRESSED_RGBA_ASTC_4x4_KHR doesn't exist in the RPI VideoCore IV GLES2 headers, and from what I have read online ASTC Texture Compression is a GLES3 extension.

hrydgard commented:
Unfortunately our old define USING_GLES2 dates back to before GLES3, so it gates that too.

I'll try to come up with something else, but I think simply

```
#ifdef GL_COMPRESSED_RGBA_ASTC_4x4_KHR 

...

#endif
```

would work, as that's a define.

joolswills commented:
Thanks. I changed the commit as you suggested as that's a better fix (and have used that in RetroPie - https://github.com/RetroPie/RetroPie-Setup/pull/3810).

joolswills committed:
Fix building on OpenGL ES 2.0 (RPI VideoCore IV)

GL_COMPRESSED_RGBA_ASTC_4x4_KHR is not defined and looks to be an OpenGL ES 3.0 extension.

This fixes building on the Raspberry Pi 1-3 using VideoCore IV OpenGL ES 2.0 headers.
